<launch>

  <group ns="calibimage">
    <arg name="std_topic" default="image_raw"/>
    
    <node pkg="image_transport" type="republish" name="$(anon repub1)" args="compressed in:=$(arg std_topic) raw out:=$(arg std_topic)" />

    <node pkg="sailing_stones" type="getpoints.py" output="screen" name="calibmon">
      <param name="source1" value="/watcher1/image_raw" />
      <param name="source2" value="/watcher2/image_raw" />
      <param name="source3" value="/watcher3/image_raw" />
      <param name="source4" value="/watcher4/image_raw" />
      <param name="source5" value="/watcher5/image_raw" />
      <param name="source6" value="/watcher6/image_raw" />
      <param name="sink1" value="/nozone1/image_raw" />
      <param name="sink2" value="/nozone2/image_raw" />
      <param name="sink3" value="/nozone3/image_raw" />
      <param name="sink4" value="/nozone4/image_raw" />
      <param name="sink5" value="/nozone5/image_raw" />
      <param name="sink6" value="/nozone6/image_raw" />
    </node>

  </group>


<group ns="calibrate">
    <arg name="std_topic" default="image_raw"/>
    


   <node pkg="image_transport" type="republish" name="$(anon repub2)" args="compressed in:=$(arg std_topic) raw out:=$(arg std_topic)" />

    <node pkg="sailing_stones" type="calibpoints.py" output="screen" name="calibpon">
      <param name="source1" value="/path1/image_raw" />
      <param name="source2" value="/path2/image_raw" />
      <param name="source3" value="/path3/image_raw" />
      <param name="source4" value="/path4/image_raw" />
      <param name="source5" value="/path5/image_raw" />
      <param name="source6" value="/path6/image_raw" />
      <param name="sink" value="/calibrated/image_raw" /> 
    </node>
  </group>
 <group ns="calibrateraw">
    <arg name="std_topic" default="image_raw"/>
    


   <node pkg="image_transport" type="republish" name="$(anon repub2)" args="compressed in:=$(arg std_topic) raw out:=$(arg std_topic)" />

    <node pkg="sailing_stones" type="calibraw.py" output="screen" name="calibraw">
      <param name="source1" value="/watcher1/image_raw" />
      <param name="source2" value="/watcher2/image_raw" />
      <param name="source3" value="/watcher3/image_raw" />
      <param name="source4" value="/watcher4/image_raw" />
      <param name="source5" value="/watcher5/image_raw" />
      <param name="source6" value="/watcher6/image_raw" />
      <param name="sink" value="/calibratedraw/image_raw" /> 
    </node>
  </group>

<groupx ns="findrobot">
    <arg name="std_topic" default="image_raw"/>
    <arg name="camera2_info_url" default="package://sailing_stones/cfg/calibcam3.yaml" />

    <node pkg="image_transport" type="republish" name="$(anon repub2)" args="compressed in:=$(arg std_topic) raw out:=$(arg std_topic)" />

    <node pkg="sailing_stones" type="findrobot.py" output="screen" name="findrobo">
      <param name="source" value="/calibrated/image_raw" /> 
    </node>


  </groupx>
</launch>
